pkg/driver/controller.go

```diff
@@ -154,9 +154,14 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 		}
 	}
 
+	snapshotID := ""
+	if snapshotSource := req.VolumeContentSource.GetSnapshot(); snapshotSource != nil {
+		snapshotID = snapshotSource.SnapshotId
+	}
+
 	// volume exists already
 	if disk != nil {
-		return newCreateVolumeResponse(disk), nil
+		return newCreateVolumeResponse(disk, snapshotID), nil
```

Comment:
I would prefer snapshot ID be part of the disk struct

Reply:
Added. But it moves us further away from https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/393